doc/api/assert.md

```diff
@@ -2548,6 +2548,241 @@ assert.throws(throwingFirst, /Second$/);
+// OK
+
+assert.deepMatch({ a: 1, b: '2', c: 3 }, { a: 1, b: 2 });
+// OK
+
+assert.deepMatch({ a: { b: { c: '1' } } }, { a: { b: { c: 1 } } });
+// OK
+
+assert.deepMatch({ a: 1 }, { a: 1, b: 2 });
+// AssertionError
+
+assert.deepMatch({ a: 1, b: true }, { a: 1, b: 'true' });
+// AssertionError
+
+assert.deepMatch({ a: { b: 2 } }, { a: { b: 2, c: 3 } });
+// AssertionError
+```
+
+```cjs
+const assert = require('node:assert');
+
+assert.deepMatch({ a: 1, b: '2' }, { a: 1, b: 2 });
+// OK
+
+assert.deepMatch({ a: 1, b: '2', c: 3 }, { a: 1, b: 2 });
+// OK
+
+assert.deepMatch({ a: { b: { c: '1' } } }, { a: { b: { c: 1 } } });
+// OK
+
+assert.deepMatch({ a: 1 }, { a: 1, b: 2 });
+// AssertionError: Expected key b
+
+assert.deepMatch({ a: 1, b: true }, { a: 1, b: 'true' });
+// AssertionError
+
+assert.deepMatch({ a: { b: 2, d: 4 } }, { a: { b: 2, c: 3 } });
+// AssertionError: Expected key c
+```
+
+If the values or keys are not equal in the `expected` parameter, an [`AssertionError`][] is thrown with a `message`
+property set equal to the value of the `message` parameter. If the `message`
+parameter is undefined, a default error message is assigned. If the `message`
+parameter is an instance of an [`Error`][] then it will be thrown instead of the
+`AssertionError`.
+
+## `assert.deepMatchStrict(actual, expected[, message])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `actual` {any}
+* `expected` {any}
+* `message` {string|Error}
+
+[`assert.deepMatchStrict()`][] Assesses the equivalence between the `actual` and `expected` parameters through a
+deep comparison, ensuring that all properties in the `expected` parameter are
+present in the `actual` parameter with equivalent values, not allowing type coercion.
+The main difference with [`assert.deepStrictEqual()`][] is that [`assert.deepMatchStrict()`][] does not require all
+properties in the `actual` parameter to be present in the `expected` parameter.
+
+```mjs
+import assert from 'node:assert';
+
+assert.deepMatchStrict({ a: 1, b: 2 }, { a: 1, b: 2 });
+// OK
+
+assert.deepMatchStrict({ a: { b: { c: 1 } } }, { a: { b: { c: 1 } } });
+// OK
+
+assert.deepMatchStrict({ a: 1, b: 2, c: 3 }, { a: 1, b: 2 });
+// OK
+
+assert.deepMatchStrict({ a: 1 }, { a: 1, b: 2 });
+// AssertionError
+
+assert.deepMatchStrict({ a: 1, b: '2' }, { a: 1, b: 2 });
+// AssertionError
+
+assert.deepMatchStrict({ a: { b: 2 } }, { a: { b: '2' } });
+// AssertionError
+```
+
+```cjs
+const assert = require('node:assert');
+
+assert.deepMatchStrict({ a: 1, b: 2 }, { a: 1, b: 2 });
+// OK
+
+assert.deepMatchStrict({ a: { b: { c: 1 } } }, { a: { b: { c: 1 } } });
+// OK
+
+assert.deepMatchStrict({ a: 1, b: 2, c: 3 }, { a: 1, b: 2 });
+// OK
+
+assert.deepMatchStrict({ a: 1 }, { a: 1, b: 2 });
+// AssertionError
+
+assert.deepMatchStrict({ a: 1, b: '2' }, { a: 1, b: 2 });
+// AssertionError
+
+assert.deepMatchStrict({ a: { b: 2 } }, { a: { b: '2' } });
+// AssertionError
+```
+
+## `assert.includes(actual, expected[, message])`
```

Comment:
I'm less convinced that this one is useful. Why not just simply use `assert(actual.includes(expected))`

Reply:
I agree with @jasnell and don't think this should be implemented. Let us concentrate on the partial inclusion. If anyone would ever ask for something else, we can still implement more.